packages/@react-spectrum/table/src/TableView.tsx

```diff
@@ -542,28 +543,43 @@ function ResizableTableColumnHeader({item, state}) {
         break;
     }
   };
+  let allowsSorting = column.props?.allowsSorting;
+  let items = useMemo(() => {
+    let options = {
+      sortAscending: allowsSorting && {
+        label: formatMessage('sortAscending'),
+        id: 'sort-asc'
+      },
+      sortDescending: allowsSorting && {
+        label: formatMessage('sortDescending'),
+        id: 'sort-desc'
+      },
+      resize: {
+        label: formatMessage('resizeColumn'),
+        id: 'resize'
+      }
+    };
+    return Object.keys(options).reduce((acc, key) => {
+      if (options[key]) {
+        acc.push(options[key]);
+      }
+      return acc;
+    }, []);
```

Comment:
Why is this written like this rather than just coding it as an array in the first place?